### DISCUSSION TITLE
Can parse JSON data into lists？

### DISCUSSION THREAD

**Discussion by touchmii** [Waboodoo/HTTP-Shortcuts#239] [Ideas]
My server wants to return some list data, can it be displayed as a list for easy scrolling。

**Comment by Waboodoo** [comment]
The closest thing to this that you can currently do with the app is to transform your list into a string, store that into a variable and display that variable. Here's roughly how that would work:

1. Open the app, click the overflow menu and select "Variables"
2. Use the + button to create a new variable of type "Static Variable (Constant)". Give it a name and save it.
3. Start editing your shortcut and go to the "Response Handling" section.
4. Change the "On Success" setting to "Show a message".
5. In the "Message" setting, click the "{}" button and select the variable you created
6. Change the "Display Type" to "Fullscreen Window"
7. Go back and into the "Scripting" section
8. In the "Run on Success" setting, enter code that parses your response and saves it into the variable. This could look something like this:

```js
const myList = JSON.parse(response.body);
const myListAsString = myList.join('\n');
setVariable('myVariable', myListAsString);
```

Once you're done with all these steps, save your shortcut and try it out.


I will keep the idea in my backlog, maybe I can improve some things in the app to make this better supported.

**Reply by Waboodoo** [reply]
Forgot to post an update here: there is actually now a feature since version 3.15.0, which is enabled by default. If the response is a JSON array of suitable structure, it will be rendered as a table.

More details here: https://http-shortcuts.rmy.ch/advanced#tables
